### PR TITLE
DDP-5973 - implement support for confirm_placeholder_template_id

### PR DESCRIPTION
**Pull request by evgeniipr**
## Context

Implement a backend support for `text_question.confirm_placeholder_template_id` (second text input for confirming text).
This is a new field (bigint) referencing to a template (table `template`). It is nullable and can contain an ID of a template with text to be shown in a confirmation input field. 
The field is added:
- into `TextQuestionDto`: `private Long confirmPlaceholderTemplateId`;
- into `TextQuestionDef`:  `private Template confirmPlaceholderTemplate`;
- into `TextQuestion`:  `private String confirmPlaceholderText`.

## Checklist

- [x] I have labeled the type of changes involved using the `C-*` labels.
- [x] I have assessed potential risks and labeled using the `R-*` labels.
- [x] I have considered error handling and alerts, and added `L-*` labels as needed.
- [x] I have considered security and privacy, and added `I-*` labels as needed
- [x] I have analyzed my changes for stability, fault tolerance, graceful degradation, performance bottlenecks and written a brief summary in this PR.
- [x] If applicable, I have discussed the analytics needs at both a platform and study level with Product and instrumented code accordingly.
- [x] If applicable, my UI/UX changes have passed muster with Product/Design via an over-the-shoulder review, screenshots, etc.

## FUD Score

_Overall, how are you feeling about these changes?_

- [x] :relaxed: All good, business as usual!
- [ ] :sweat_smile: There might be some issues here
- [ ] :scream: I'm sweaty and nervous

## How do we demo these changes?

_How does one observe these changes in a deployed system? Note that **user visible** encompasses many personas--not just patients and study staff, but also ops duty, your fellow devs, compliance, etc._

- [x] They are user-visible in dev as a regular user journey and require no additional instructions.
- [ ] Getting dev into a state where this is user-visible requires some tech fiddling. I have documented these steps in the related ticket.
- [ ] Requires other features before it's human visible. I have documented the blocking issues in jira.
- [ ] I have no idea how to demo this. Please help me!

## Testing

- [x] I have written automated positive tests
- [ ] I have written automated negative tests
- [ ] I have written zero automated tests but have poked around locally to verify proper functionality
- [ ] The jira ticket has acceptance criteria and QA has the needed information to test changes

## Release

- [ ] These changes require no special release procedures--just code!
- [ ] Releasing these changes requires special handling and I have documented the special procedures in the release plan document

### CODE REVIEW

File: pepper-apis/src/main/java/org/broadinstitute/ddp/util/TemplateRenderUtil.java [hunk 1]
```diff
@@ -18,4 +20,18 @@ public static String toPlainText(Long templateId, Renderable.Provider<String> re
         }
         return renderedValue;
     }
+
+    public static String applyRenderedTemplate(Long templateId, String templateText,
+                                               Renderable.Provider<String> rendered, ContentStyle style) {
+        if (templateId != null) {
+            templateText = rendered.get(templateId);
+            if (templateText == null) {
+                throw new NoSuchElementException("No rendered template found for placeholder with id " + templateId);
```
Comment (yufengwng) [inline pepper-apis/src/main/java/org/broadinstitute/ddp/util/TemplateRenderUtil.java]: `templateId` passed in might not be a placeholder...

File: pepper-apis/src/main/java/org/broadinstitute/ddp/db/dao/JdbiTextQuestion.java [hunk 2]
```diff
@@ -31,11 +33,13 @@ int insert(@Bind("questionId") long questionId,
...
-            + "     tq.placeholder_template_id = :placeholderTemplateId "
+            + "     tq.placeholder_template_id = :placeholderTemplateId, "
+            + "     tq.confirm_placeholder_template_id = :confirmPlaceholderTemplateId "
             + "WHERE tq.question_id = :questionId")
     boolean update(@Bind("questionId") long questionId,
                    @Bind("inputType") TextInputType inputType,
                    @Bind("suggestionType") SuggestionType suggestionType,
-                   @Bind("placeholderTemplateId") Long placeholderTemplateId);
+                   @Bind("placeholderTemplateId") Long placeholderTemplateId,
+                   @Bind("confirmPlaceholderTemplateId") Long confirmPlaceholderTemplateId);
```
Comment (yufengwng) [inline pepper-apis/src/main/java/org/broadinstitute/ddp/db/dao/JdbiTextQuestion.java]: Confirm placeholder goes along with confirmEntry, confirmPrompt, and mismatchMessage. Does it make sense to allow updating `confirmPlaceholderTemplateId` when  the other "confirm" properties are not part of the method? I suggest removing `confirmPlaceholderTemplateId` from here.

File: pepper-apis/src/main/java/org/broadinstitute/ddp/model/activity/instance/question/TextQuestion.java [hunk 3]
```diff
@@ -82,34 +88,12 @@ public TextQuestion(String stableId, long promptTemplateId, Long placeholderTemp
...
-                inputType,
-                suggestionType,
-                suggestions,
-                confirmEntry,
-                confirmPromptTemplateId,
-                mismatchMessageTemplateId);
-    }
-
-    public TextQuestion(String stableId, long promptTemplateId, Long placeholderTemplateId,
+    public TextQuestion(String stableId, long promptTemplateId, Long placeholderTemplateId, Long confirmPlaceholderTemplateId,
```
Comment (yufengwng) [inline pepper-apis/src/main/java/org/broadinstitute/ddp/model/activity/instance/question/TextQuestion.java]: Does it make sense to pass in `confirmPlaceholderTemplateId` when we're not passing in other properties like confirmEntry, confirmPrompt, etc? I think it'll make more sense to add `confirmPlaceholderTemplateId` to the constructor that was deleted above.

**Review by yufengwng — COMMENTED** [review]
Looks good, just a small question.

File: pepper-apis/src/main/java/org/broadinstitute/ddp/util/TemplateRenderUtil.java (hunk 1 above)
Comment (evgeniipr) [reply pepper-apis/src/main/java/org/broadinstitute/ddp/util/TemplateRenderUtil.java]: Fixed

File: pepper-apis/src/main/java/org/broadinstitute/ddp/db/dao/JdbiTextQuestion.java (hunk 2 above)
Comment (evgeniipr) [reply pepper-apis/src/main/java/org/broadinstitute/ddp/db/dao/JdbiTextQuestion.java]: Removed

File: pepper-apis/src/main/java/org/broadinstitute/ddp/model/activity/instance/question/TextQuestion.java (hunk 3 above)
Comment (evgeniipr) [reply pepper-apis/src/main/java/org/broadinstitute/ddp/model/activity/instance/question/TextQuestion.java]: But in this constructor other confirm-ids also passed:. This version from develop:
```
public TextQuestion(String stableId, long promptTemplateId, Long placeholderTemplateId, List<TextAnswer> answers,
                        List<Rule<TextAnswer>> validations, TextInputType inputType, SuggestionType suggestionType,
                        List<String> suggestions, boolean confirmEntry,
                        Long confirmPromptTemplateId, Long mismatchMessageTemplateId)
```

Comment (evgeniipr) [reply pepper-apis/src/main/java/org/broadinstitute/ddp/model/activity/instance/question/TextQuestion.java]: I can return back that 1st constructor. But it is not used anywhere.

Comment (yufengwng) [reply pepper-apis/src/main/java/org/broadinstitute/ddp/model/activity/instance/question/TextQuestion.java]: For this constructor I'm looking at (line 91)
```java
public TextQuestion(String stableId, long promptTemplateId, Long placeholderTemplateId, Long confirmPlaceholderTemplateId,
                    List<TextAnswer> answers, List<Rule<TextAnswer>> validations, TextInputType inputType) {
```
why is confirmPlaceholder there when other properties like confirmEntry, confirmPrompt, etc are not?

Comment (yufengwng) [reply pepper-apis/src/main/java/org/broadinstitute/ddp/model/activity/instance/question/TextQuestion.java]: The constructor you posted from develop is not the one I'm talking about. That one is fine and we should add confirmPlaceholder to it. But this one I'm pointing out shouldn't have confirmPlaceholder -- it doesn't make sense here unless we also pass in other "confirm" properties too.

Comment (evgeniipr) [reply pepper-apis/src/main/java/org/broadinstitute/ddp/model/activity/instance/question/TextQuestion.java]: Ah, sorry.. Fixed.

**Comment by yufengwng** [comment]
Can you also update the OpenAPI documentation to include this new property?

**Comment by evgeniipr** [comment]
> Can you also update the OpenAPI documentation to include this new property?

I added in schemas.yml:
```
 confirmPlaceholderText:
        type: string
        nullable: true
```
or do you mean some description of this field?

**Review by yufengwng — APPROVED** [review]
Ah yes, I see it! I must have missed it earlier.

**Pull request merged**